dali/operators/generic/histogram/histogram.cc

```diff
@@ -0,0 +1,685 @@
+                                 nullptr);
+
+DALI_SCHEMA(HistogramOpName)
+    .DocStr(R"code("Calculates 1D or ND histogram of the input tensor with non-uniform histogram bin ranges.
+
+This version of histogram function in contrast to ``UniformHistogram`` requires specifing tensor(s) of histogram bin ranges.
+Ranges of histogram must be specified as tensor number of bins plus one float32 elements.
+The numbers specify bin boundries such that, first of them is bin inclusive lower bound and the second is exclusive upper bound and in same time lower bound of next bin.
+
+For example range tensor with sequence:
+
+``A, B, C, D``
+
+would describe 3 bins ``[A, B) [B, C), [C, D)``
+
+When calulating multidimensional (see below) ranges should be specified as an additional tensor argument for each dimension.
+
+The histogram of the input tensor when ``channel_axis`` or ``channel_axis_name`` is specified is calculated as multidimensional
+histogram ie. as if histogram would be calculated for each seperate channel of this axis.
+If channel axis is not specified 1D histogram is calculated.
+Current implentation supports up to 32 channels for histogram calculation.
+
+Histogram calculation supports specifing arbitrary axes of reduction.
+
+For example for tensor with layout "HWC" one could calculate different single and multidimentional histograms.
+
+One could specify ``axes_names = "W"`` as an reduction axis tensor of "HC" histograms in dimension "W" would be calculated.
+By not specifing reduction axis all axes are taken of consideration and single dimesional histogram of all axes is calculated.
+    )code")
+    .NumInput(2, 1 + HISTOGRAM_MAX_CH)
+    .NumOutput(1)
+    .AddParent("HistogramBase");
+
+DALI_SCHEMA(UniformHistogramOpName)
+    .DocStr(R"code(Calculates 1D or ND histogram of the input tensor with uniform histogram bin ranges.
+
+Calculates histogram of of uniform bin ranges, second input tensor specifies lower bound of range of values and third argument specfies
+upper range of values in each histogram dimension.
+
+For example lower range (2nd tensor argument) ``[0]`` and upper range (3rd tensor argument) ``[255]`` and ``num_bins=[16]`` will calculate histogram
+with 16 bins of uniformly subdivided in range ``[0, 255]``
+
+For higher dimensional histogram each ``num_bins``, lower and upper range input tensor element describe such subdivision in corresponding dimension.
+
+The histogram of the input tensor when ``channel_axis`` or ``channel_axis_name`` is specified is calculated as multidimensional
+histogram ie. as if histogram would be calculated for each seperate channel of this axis.
+If channel axis is not specified 1D histogram is calculated.
+Current implentation supports up to 32 channels for histogram calculation.
+
+Histogram calculation supports specifing arbitrary axes of reduction.
+
+For example for tensor with layout "HWC" one could calculate different single and multidimentional histograms.
```

Comment:
I posted some comments in the other occurrence of this text. I think those comments apply here as well